### PR TITLE
Fix: address validation

### LINKED ISSUE #20666: "It's not Ethereum address or ENS" validation error shown for a second when pasting a valid address into "Send to" field
# Bug Report

### Reproduction

1. Copy any valid address
2. Open wallet 
3. Open an account > press `Send` 
4. Paste the address or scan a QR code with a valid address
5. Pay attention to the validation error

#### Expected behavior

No error

#### Actual behavior

https://github.com/status-im/status-mobile/assets/67952253/6c33853f-93d3-4dd1-b7a3-652c523a2cec

### Additional Information

- Status version: nightly 5/07
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

### PR DESCRIPTION
**Pull request by OmarBasem**
fixes: https://github.com/status-im/status-mobile/issues/20666

### Summary
This PR fixes validation error shown for a second when pasting a valid address

### Demo

https://github.com/status-im/status-mobile/assets/29354102/e6382cdb-a17c-423f-bd31-ba87167b3aee

### CODE REVIEW

File: src/status_im/subs/wallet/wallet.cljs [hunk 1]
```diff
@@ -595,6 +595,11 @@
  :<- [:wallet/search-address]
  :-> :valid-ens-or-address?)
 
+(rf/reg-sub
+ :wallet/searching-address?
+ :<- [:wallet/search-address]
+ :-> :loading?)
+
```
Comment (OmarBasem) [inline src/status_im/subs/wallet/wallet.cljs:602]: Added a `loading?` key to the `search-address` to know if we are ready to show errors if any or not

File: src/status_im/contexts/wallet/events.cljs [hunk 2]
```diff
@@ -362,18 +362,26 @@
 
 (rf/reg-event-fx :wallet/address-validation-success
  (fn [{:keys [db]}]
-   {:db (assoc-in db [:wallet :ui :search-address :valid-ens-or-address?] true)}))
+   {:db (-> db
+            (assoc-in [:wallet :ui :search-address :valid-ens-or-address?] true)
+            (assoc-in [:wallet :ui :search-address :loading?] false))}))
```
Comment (BalogunofAfrica) [inline src/status_im/contexts/wallet/events.cljs]: This updates the properties within a nested keyword ([:wallet :ui :search-address]) in the db in one step

```suggestion
   {:db (update-in db
                     [:wallet :ui :search-address]
                     assoc
                     :valid-ens-or-address?     true
                     :loading?                  false)}
```

Comment (OmarBasem) [reply src/status_im/contexts/wallet/events.cljs]: Thanks for the suggestion @BalogunofAfrica, updated it!

**Review by BalogunofAfrica — APPROVED** [review]

File: src/status_im/contexts/wallet/events.cljs (hunk 2 above)
Comment (ulisesmac) [reply src/status_im/contexts/wallet/events.cljs]: @BalogunofAfrica in this case I think we don't have a strong preference, sometimes when the code is easier to understand or less verbose it's fine, in this case I think it's the same, but it's always good to share different approaches to get the same result :100:

**Comment by mariia-skrypnyk** [comment]
Hi @OmarBasem !

Thanks for your fix.
Looks good on both platforms.
Failed e2e are known issue caused by Waku.
You can merge your PR.

**Pull request merged**